core/app_modules/cal.js

```diff
@@ -7,7 +7,17 @@ module.exports.addCalendarEvent = co.wrap(function*(username, event, reply){
   var client = db();
   var a = yield auth.isAdmin(username);
   var user = a ? event.user : reply;
-  client.query('INSERT INTO calendar("user", "time", room, duration, title)VALUES ($1, $2, $3, $4, $5);', [user, event.time, event.room, event.duration, event.title]);
+  
+  var isOverlap = false;
+  var allCalendarEvents = yield client.query("SELECT * FROM calendar;");
+  for (var i = 0; i < allCalendarEvents; i++) {
+    if ((event.time >= allCalendarEvents[i].time && event.time <= allCalendarEvents[i].time + allCalendarEvents[i].duration) && (event.time+event.duration >= allCalendarEvents[i].time && event.time+event.duration <= allCalendarEvents[i].time+allCalendarEvents[i].duration)) {
+      isOverlap = true;
+    }
+  }
+  if (!isOverlap) {
+    client.query('INSERT INTO calendar("user", "time", room, duration, title)VALUES ($1, $2, $3, $4, $5);', [user, event.time, event.room, event.duration, event.title]);
+  }

```

Comment:
Nicely done! Do you want to reject the promise if there's an overlap and then add a test for that contingency?